x-pack/solutions/observability/plugins/observability_agent/server/tools/get_change_points/get_metric_change_points.ts

```diff
@@ -0,0 +1,304 @@
+      terms: groupingFields.map((groupingField) => ({ field: groupingField })),
+      size: 10,
+    },
+  };
+}
+
+async function getMetricChangePoints({
+  name,
+  index,
+  start,
+  end,
+  kqlFilter,
+  groupBy,
+  field,
+  type,
+  esClient,
+}: {
+  name: string;
+  index: string;
+  start: string;
+  end: string;
+  kqlFilter?: string;
+  groupBy: string[];
+  field?: string;
+  type: MetricType;
+  esClient: IScopedClusterClient;
+}) {
+  const { agg: metricAgg, buckets_path: metricBucketsPath } = getMetricAggregation({
+    type,
+    field,
+  });
+
+  const groupAgg = getGroupingAggregation(groupBy);
+
+  const subAggs = {
+    over_time: {
+      auto_date_histogram: dateHistogram,
+      aggs: {
+        metric: metricAgg,
+        value: {
+          bucket_script: {
+            buckets_path: {
+              metric: `metric${metricBucketsPath ? `>${metricBucketsPath}` : ''}`,
+            },
+            script: 'params.metric',
+          },
+        },
+      },
+    },
+    changes: {
+      change_point: {
+        buckets_path: 'over_time>value',
+      },
+      // elasticsearch@9.0.0 change_point aggregation is missing in the types: https://github.com/elastic/elasticsearch-specification/issues/3671
+    } as AggregationsAggregationContainer,
+  };
+
+  const aggregations = {
+    groups: {
+      ...groupAgg,
+      aggs: subAggs,
+    },
+  };
+
+  const response = await searchChangePoints({
+    esClient,
+    index,
+    start,
+    end,
+    kqlFilter,
+    aggregations,
+  });
+
+  const buckets = (response.aggregations as AggregationsResponse | undefined)?.groups?.buckets;
+  if (!buckets) {
+    return [];
+  }
+
+  return await getChangePoints({
+    name,
+    buckets,
+  });
+}
+
+const getMetricChangePointsSchema = z.object({
+  start: z
+    .string()
+    .describe(
+      'The beginning of the time range, in Elasticsearch datemath, like `now-24h`, or an ISO timestamp'
+    ),
+  end: z
+    .string()
+    .describe(
+      'The end of the time range, in Elasticsearch datemath, like `now`, or an ISO timestamp'
+    ),
+  metrics: z
+    .array(
+      z.object({
+        name: z.string().describe('The name of the set of metrics'),
+        index: z.string().describe('The index or index pattern to find the metrics').optional(),
+        kqlFilter: z
+          .string()
+          .describe('A KQL filter to filter the metric documents, e.g.: my_field:foo')
+          .optional(),
+        field: z
+          .string()
+          .describe(
+            'Metric field that contains the metric. Only use if the metric aggregation type is not `count`.'
+          )
+          .optional(),
+        type: z
+          .enum(['count', 'avg', 'sum', 'min', 'max', 'p95', 'p99'])
+          .describe('The type of metric aggregation to perform. Defaults to `count`')
+          .optional(),
+        groupBy: z
+          .array(z.string())
+          .describe('Optional keyword fields to group metrics by.')
```

Comment:
What about adding known fields that are common to group by as examples? I think the most common or valuable fields to group by are `service.name`, `service.environment`, `service.version`, `event.outcome`, `log.level`, `event.dataset`, `host.name`

```suggestion
          .describe('Optional keyword fields to group metrics by.')
```